### PR TITLE
🧬 Oasis Create table

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# Oasis Create table

Covers:
https://app.shortcut.com/oazo-apps/story/9002/update-assets-table-layout
https://app.shortcut.com/oazo-apps/story/8915/update-filter-dropdown
https://app.shortcut.com/oazo-apps/story/8918/add-filtering-for-oasis-create

This PR in reality is smaller than it seems. Some of the changes are new svg files or edits to existing ones and there is also a 400 lines file `oasisCreate/data.ts` that contains dummy data as a source for the table, that should be removed in the future and replaced with API call.

image
  
## Changes 👷‍♀️

- Renamed `DiscoverMultiselect` to `GenericMultiselect`, updated its design and extended functionality - for some reason Github is showing entire file as new 😒,
- updated design of `AssetsTable` and its dependencies to match new layout,
- updated design of `GenericSelect` external plugin wrapper,
- updated design of `HeaderSelector`,
- redid some of Discover filters to make them more generic and usable also on Oasis Create page,
- changed layout background for Discover and owner's page,
- updated `WithArrow` component so it uses `span` instead of `div` and we can get rid of constant error messages in console about using `div` as a children of `p`,
   
### Before you start testing:
- turn on `OasisCreate` feature flag,
- data is the table is fake, but filtering works,
- for the purpose of sorting, n/a means 0,
- `8.6%` duplication rate is only because of the dummy data file, rest of the files doesn't have any duplication.

### Oasis Create page
Address of this page is changing dynamically, so you can reach it by different routes:
- `oasis-create/borrow`
- `oasis-create/multiply/WBTC`
of course, the final product is going to be available as oasis.app/multiply/WBTC

### Owner's page
This page design changed along with table update, please take a look if you're okay with how it looks in new design and if I need to change something. couple addresses with diversified portfolio for testing:
- `/owner/0xd0612c759fC6C512B7C3abceB2366C4f4096451C`.
- `/owner/0x165d0df29d9262f8ee9e9874c906eab706a05c2c`.
- `/owner/0x10649c79428d718621821Cf6299e91920284743F`.

### CODE REVIEW

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: features/discover/layout.tsx [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import { PageSEOTags } from 'components/HeadTags'
-import { MarketingLayout } from 'components/Layouts'
+import { AppLayout } from 'components/Layouts'
 import { INTERNAL_LINKS } from 'helpers/applicationLinks'
+import { WithChildren } from 'helpers/types'
 import React from 'react'
 
-export const discoverPageLayout = MarketingLayout
+export const discoverPageLayout = ({ children }: WithChildren) => (
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply features/discover/layout.tsx]: Fixed.

File: pages/oasis-create/[...slug].tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    locales
+      ?.map((locale) =>
+        Object.values(ProductType)
+          .map((product) =>
+            Object.values(oasisCreateOptionsMap[product].tokens).map((token) => [
+              product,
+              ...(token.value !== ALL_ASSETS ? [token.value] : []),
+            ]),
+          )
+          .flat()
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply pages/oasis-create/[...slug].tsx]: Fixed.

File: features/oasisCreate/views/OasisCreateView.tsx [hunk 3]
```diff
@@ -91,9 +131,63 @@ export function OasisCreateView({ product }: OasisCreateViewProps) {
...
+          <AssetsResponsiveTable
+            rows={parsedRows}
+            headerTranslationProps={{
+              ...(selectedToken && { token: selectedToken === ALL_ASSETS ? 'ETH' : selectedToken }),
+            }}
+          />
+        ) : (
+          <AssetsTableNoResults
+            header="There are no items matching your filters"
+            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean iaculis lorem in feugiat mattis."
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply features/oasisCreate/views/OasisCreateView.tsx:189]: Fixed.

File: components/GenericMultiselect.tsx [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+        {icon && <Icon size={32} name={icon} sx={{ verticalAlign: 'bottom' }} />}
+        {image && (
+          <Image
+            src={image}
+            alt={label}
+            sx={{ width: '26px', height: '26px', verticalAlign: 'bottom' }}
+          />
+        )}
+      </Box>
+    </>
```
Comment (marcinciarka) [inline components/GenericMultiselect.tsx:45]: Unnecessary fragment.

File: components/GenericMultiselect.tsx [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+    if (didMountRef.current) onChange(values.length ? values : options.map((item) => item.value))
+    else didMountRef.current = true
+  }, [values])
+
+  function getSelectLabel(): ReactNode {
+    switch (values.length) {
+      case 0:
+        return (
+          <>
+            {icon && <Icon name={icon} size={32} sx={{ flexShrink: 0, mr: '12px' }} />}
```
Comment (marcinciarka) [inline components/GenericMultiselect.tsx:158]: If this is 32px, it should be "32px"

**Review by marcinciarka — DISMISSED** [review]

File: components/GenericMultiselect.tsx (hunk 5 above)
Comment (piotrkonowrocki) [reply components/GenericMultiselect.tsx:158]: `size` expects `number`

**Review by marcinciarka — APPROVED** [review]

**Review by piekczyk — APPROVED** [review]

**Pull request merged**